admin/src/components/Comment/index.js

```diff
@@ -0,0 +1,137 @@
+import { h, Component } from 'preact';
+import axios from 'axios';
+import Button from '../Button';
+
+export default class Comment extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      author: this.props.author,
+      body: this.props.body,
+    };
+
+    this.updateComment = this.updateComment.bind(this);
+    this.deleteComment = this.deleteComment.bind(this);
+    this.undoDeleteComment = this.undoDeleteComment.bind(this);
+    this.confirmComment = this.confirmComment.bind(this);
+    this.handleAuthorChange = this.handleAuthorChange.bind(this);
+    this.handleBodyChange = this.handleBodyChange.bind(this);
+  }
+
+  handleAuthorChange(event) {
+    this.setState({ author: event.target.value });
+  }
+
+  handleBodyChange(event) {
+    this.setState({ body: event.target.value });
+  }
+
+  async updateComment() {
+    try {
+      const data = {
+        CommentId: this.props.id,
+        Body: this.props.body,
+        Author: this.props.author
+      };
+      const res = await axios.patch(`${window.location.origin}/v1/admin/comments`, data);
+
+      if (res.status === 204) {
+        // TODO: reload comments
+        console.log('Hooray! Comment updating successful');
+      }
+    } catch (err) {
+      console.log('Error while updating comment', err);
+    }
+  }
+
+  async confirmComment() {
+    // NOTE: dunno what should I think about this
+    try {
+      const data = {
+        // CommentId: this.props.id,
+        // Body: this.props.body,
+        // Author: this.props.author,
+        Confirmed: true,
```

Comment:
do we also need to update the comment state. We don't need to show the button if the comment is already confirmed.